docs/designers-developers/developers/block-api/block-deprecation.md

```diff
@@ -7,12 +7,18 @@ When updating static blocks markup and attributes, block authors need to conside
 
 A block can have several deprecated versions. A deprecation will be tried if a parsed block appears to be invalid, or if there is a deprecation defined for which its `isEligible` property function returns true.
 
+Deprecations do not operate as a chain of updates in the way other software data updates, like database migrations, do. At first glance, it is easy to think that each deprecation is going to make the required changes to the data and then hand this new form of the block onto the next deprecation to make its changes. What happens is that each deprecation is passed the original saved content, and if its `save` method produces a valid block it is used to parse the block attributes. If it has a `migrate` method it will also be run, and the data is then passed back to the current block `save` function to create the new valid version of the block.
+
+It is also important to note that if a deprecation's `save` method does not produce a valid block then it is skipped, including its `migrate` method, even if `isEligible` returns true. This means that if you have several deprecations for a block and want to perform a new migration, like moving content to InnerBlocks, you may need to include the `migrate` method in multiple deprecations for it to be applied to all previous versions of the block.
```

Comment:
my understanding is that the deprecations are run if the current `save` produces an invalid block, regardless of the `isEligible` return ... but if the current `save` is valid then `isEligible` is called to see if the deprecation should run anyway:
```
if ( block.isValid && ! isEligible( parsedAttributes, innerBlocks ) ) {
    continue;
}
```
but I may be misunderstanding it, would be good to get this clarified.

Reply:
> my understanding is that the deprecations are run if the current save produces an invalid block, regardless of the isEligible return ... but if the current save is valid then isEligible is called to see if the deprecation should run anyway:

If I understand the `getMigratedBlock` function correctly, the `block.isValid` property is only set to true once a deprecation is successfully applied. So it is not, if the "current save" produces an invalid block but rather if after the deprecations processed so far still leave the block in an invalid state.

Your point that `isEligible` is not executed in every circumstance, is correct. To me the `isEligible` provides the means to migrate an otherwise valid block.